File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.go
func twoSum(numbers []int, target int) []int {
    var n int = len(numbers)
    var L int = 0
    var R int = n -1
    var h = []int{}
    
    for L < R{
        res := numbers[L] + numbers[R]
        
        if res == target{
            h = append(h, L+1, R+1)
            return h
        } else if res < target{
            L++
        } else if res > target{
            R--
        }
    }
    h = append(h, -1,-1)
    return h
     
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        n = len(numbers)
        L = 0
        R = n -1
        
        while (L < R):
            res = numbers[L] + numbers[R]
            
            if res == target:
                return [L+1, R+1]
            elif res < target:
                L += 1
            elif res > target:
                R -= 1
            else:
                return [-1,-1]